lib/mocha/any_instance_method.rb

```diff
@@ -13,8 +13,10 @@ def reset_mocha
 
     def restore_original_method
       return if use_prepended_module_for_stub_method?
-      return unless stub_method_overwrites_original_method?
-      original_method_owner.send(:define_method, method_name, original_method)
+      if stub_method_overwrites_original_method?
+        original_method_owner.send(:define_method, method_name, original_method)
+      end
+      return unless original_visibility
```

Comment:
Can you explain whether these changes are a pure refactoring or do they potentially change the behaviour even if they don't break any tests? Either way it would be helpful to expand the commit message to explain a bit more about how you are able to make this change with confidence!